linux_docker_resources/Dockerfile

```diff
@@ -51,6 +51,7 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
   libxml2-dev \
   libxml2-utils \
   libxslt-dev \
+  libyaml-cpp-dev \
```

Comment:
Should we also install `yaml-cpp-devel` on RHEL?

Reply:
The RHEL package list is missing a bunch of dependencies. I have a test build going that includes all of the missing deps, and I'll make a separate PR.

Reply:
Will be addressed as part of #655.